### PR TITLE
[subprocess_output] Fix exception raised on empty output + string cmd

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Without this fix, `_util.get_subprocess_output` wouldn't raise an exception
on empty output, and would instead return `None` (which would then raise
a `TypeError` since we try to use `None` as a tuple).

Fix that.

Also, fix typo for the string cmd logic.

### CODE REVIEW

File: pkg/collector/py/datadog_agent.c [hunk 1]
```diff
@@ -86,7 +86,7 @@ static PyObject *get_subprocess_output(PyObject *self, PyObject *args) {
         if (subprocess_arg == NULL) {
             PyErr_SetString(PyExc_Exception, "unable to parse arguments to cgo/go-land");
             free(subprocess_args);
```
Comment (truthbk) [inline pkg/collector/py/datadog_agent.c]: we're not releasing the GIL

**Review by truthbk — COMMENTED** [review]
Thanks for the cleanup. We've got an unreleased GIL, I think.

File: pkg/collector/py/datadog_agent.c (hunk 1 above)
Comment (olivielpeau) [reply pkg/collector/py/datadog_agent.c]: fixed 👍

**Review by truthbk — APPROVED** [review]
🚀

**Pull request merged**